pkg/webhook/pod/pod_mutater.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pod
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/BurntSushi/toml"
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/controller"
+	"github.com/pingcap/tidb-operator/pkg/features"
+	"github.com/pingcap/tidb-operator/pkg/label"
+	operatorUtils "github.com/pingcap/tidb-operator/pkg/util"
+	"github.com/pingcap/tidb-operator/pkg/webhook/util"
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+)
+
+func (pc *PodAdmissionControl) mutatePod(ar *admissionv1beta1.AdmissionRequest) *admissionv1beta1.AdmissionResponse {
+	pod := &corev1.Pod{}
+	if err := json.Unmarshal(ar.Object.Raw, pod); err != nil {
+		return util.ARFail(err)
+	}
+	original := pod.DeepCopy()
+	l := label.Label(pod.Labels)
+	if !l.IsManagedByTiDBOperator() {
+		return util.ARSuccess()
+	}
+	if !l.IsTiKV() {
+		return util.ARSuccess()
+	}
+	tcName, exist := pod.Labels[label.InstanceLabelKey]
+	if !exist {
+		return util.ARSuccess()
+	}
+	namespace := ar.Namespace
+
+	tc, err := pc.operatorCli.PingcapV1alpha1().TidbClusters(namespace).Get(tcName, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return util.ARSuccess()
+		}
+		return util.ARFail(err)
+	}
+
+	if features.DefaultFeatureGate.Enabled(features.AutoScaling) {
+		err := pc.tikvHotRegionSchedule(tc, pod)
+		if err != nil {
+			return util.ARFail(err)
+		}
+	}
+
+	patch, err := util.CreateJsonPatch(original, pod)
+	if err != nil {
+		return util.ARFail(err)
+	}
+	return util.ARPatch(patch)
+}
+
+func (pc *PodAdmissionControl) tikvHotRegionSchedule(tc *v1alpha1.TidbCluster, pod *corev1.Pod) error {
+	podName := pod.Name
+	ordinal, err := operatorUtils.GetOrdinalFromPodName(podName)
+	if err != nil {
+		return err
+	}
+	sets := operatorUtils.GetAutoScalingOutSlots(tc, v1alpha1.TiKVMemberType)
+	if !sets.Has(ordinal) {
+		return nil
+	}
+
+	cmName := controller.TiKVMemberName(tc.Name)
+	cm, err := pc.kubeCli.CoreV1().ConfigMaps(tc.Namespace).Get(cmName, metav1.GetOptions{})
```

Comment:
What if the `ConfigUpdateStrategyRollingUpdate` is set for the configmap?

Reply:
fixed in https://github.com/pingcap/tidb-operator/pull/1871